sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -53,7 +53,7 @@ trait V2WriteCommand extends UnaryCommand {
           val outType = CharVarcharUtils.getRawType(outAttr.metadata).getOrElse(outAttr.dataType)
           // names and types must match, nullability must be compatible
           inAttr.name == outAttr.name &&
```

Comment:
shall we be consistent and also respect case sensitivity here?

Reply:
It's actually more complicated than it looks. We have a rule `ResolveOutputRelation` to do the by-name resolution for top-level columns. Case sensitivity is not the only inconsistency between top-level columns and inner fields. We should update `ResolveOutputRelation` to take care of inner fields as well.